### DISCUSSION TITLE
Tutorial Code Taking 12+ hours to run

### DISCUSSION THREAD

**Discussion by will-arden** [larq/larq#877] [Q&A]
I have tried out the tutorial code for the BNN using IDLE. It causes the application to crash and go into "Not Responding" unless the window is maximized and nothing else clicked. When I do let it run, I left it on all night and into the morning - still not complete. This contrasts with the suggestion in the tutorial that it will take "a few minutes" depending on hardware. I am using a powerful laptop with a Ryzen 7 core. Any help would be appreciated!
![image](https://github.com/larq/larq/assets/31668269/74c0bdc3-97ee-4e7e-8ef1-90dd44f82e9d)

**Comment by lgeiger** [comment]
> It causes the application to crash and go into "Not Responding" unless the window is maximized and nothing else clicked.

I'm unfamiliar with IDLE but from your screenshot it seems like the terminal isn't able to correctly handle the Keras progress bar. 
Can you set `verbose=2` when calling `model.fit()` and see whether this fixes your problem? Setting `verbose=2` should disable the progress bar such that only one line per epoch is logged.
